src/swap/components/SwapProvider.tsx

```diff
@@ -86,9 +91,17 @@ export function SwapProvider({
   const [hasHandledSuccess, setHasHandledSuccess] = useState(false);
   const { from, to } = useFromTo(address);
   const { sendTransactionAsync } = useSendTransaction(); // Sending the transaction (and approval, if applicable)
+  const { sendCallsAsync } = useSendCalls(); // Atomic Batch transactions (and approval, if applicable)
```

Comment:
In what situations is `approval` applicable?

Reply:
when `from` is ERC-20

Reply:
the approval can be in the batch, but not all batches will contain approvals